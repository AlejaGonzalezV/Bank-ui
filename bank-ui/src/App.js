import React from 'react';
import './App.css';
import Navbar from './components/Navbar';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Home from './Pages/Home';
import { makeStyles } from '@material-ui/core/styles';
import AddUser from './Pages/Add-user';
import EditUser from './Pages/Edit-user';

const useStyles = makeStyles((theme) => ({
  Centerbar: {
    paddingLeft:'500'
  },
 
}));

function App() {
  const classes = useStyles();
  return (
    <>
      <Router>
        <Navbar className={classes.Centerbar} />
        <Switch>
          <Route  path='/' exact component={Home} />
          <Route path='/register'  component={AddUser}/>
          <Route path='/edit'  component={EditUser}/>

        </Switch>
      </Router>
    </>
  );
}

export default App;
