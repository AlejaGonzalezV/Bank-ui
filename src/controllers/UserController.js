import axios from 'axios';

export const UserController = {
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: '/users'
        });
    },
    register(user) {
        return axios({
            url: `users`,
            baseURL: process.env.API,
            data: user,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
              },
        })
    },
    delete(id) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `users/${id}`,
        });
    },
    edit(user) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `users/${user.document}`,
            data: user,
            headers: {
                'Content-Type': 'application/json',
              },
        });
    },
}
