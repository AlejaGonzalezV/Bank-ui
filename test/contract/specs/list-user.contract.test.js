import { provider } from "./init-pact";
import { UserController } from "../../../src/controllers";
import { Matchers } from "@pact-foundation/pact";

describe('Given An User service', () => {
    describe('When a request to list all the users is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'has users',
                uponReceiving: 'a request to get all users',
                withRequest: {
                    method: 'GET',
                    path: '/users'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike(
                        {
                        name: Matchers.string("John Doe"),
                        document: Matchers.like(1111),
                        id: Matchers.like(11),
                        username: Matchers.string("Jd12"),
                        active: Matchers.boolean(true)
                        }
                    )
                }
            });
        });

        it("Then it should return the right data", async() =>{
            const response = await UserController.list();
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
});