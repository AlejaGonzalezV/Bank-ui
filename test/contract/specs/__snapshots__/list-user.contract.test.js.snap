// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An User service When a request to list all the users is made Then it should return the right data 1`] = `
Array [
  Object {
    "active": true,
    "document": 1007554029,
    "id": 2,
    "name": "Jota Mario",
    "username": "jota123",
  },
  Object {
    "active": true,
    "document": 44444,
    "id": 61,
    "name": "asas",
    "username": "asas",
  },
]
`;
